### PR TITLE
Organ Germiness Adjustments

### PR DESCRIPTION
**Pull request by Crazylemon64**
This patch does the following:

* Organs on the same tile as an open freezer won't get more infected
* Organs will now rot slower at first, if clean enough
* Internal organ manipulation surgery will now get germs on the organs you twiddled with
* Cracking open an organ that's off of a body will spread germs to the parts involved
* Organs inside organs will now be properly preserved, if in a freezer
* Spaceacillin is 4x as strong
* Medical starts out with twice as much of it

This should address #2854, and also infections being p. much space cancer

I tried to have MMIs actually store the brain, but when I did that Cthulhu came along and knocked me out (read: I lost connection whenever I tried to view the brain in View Variables).
It'll still work otherwise, but I suspect this reflects top-notch eldritch jibberjabber that probably forebodes really really bad things, so you prolly don't want to merge this quite yet.
* MMIs now actually store the brain inside the object, so that it doesn't magically repair brains placed within
* Brains stored within an MMI will now slowly reduce germ level, so that brains left outside of one aren't doomed with this fix
* Rotten brains no longer work in an MMI (This might be harsh, should I leave this out?)

Any advice?

### CODE REVIEW

**Comment by Fox-McCloud** [comment]
Glad there's fixes,but I really hate germs--in general, it just fucks players over so badly with them having literally no way of remedying it.

Have an infection in an internal organ? Good luck ever getting it cured--you'll end up using up the entirety of medbay's spaceacillin on a single patient and it's likely to spread to other organs, in the process, while you're waiting for it to cure.

**Comment by Crazylemon64** [comment]
Well, as it stands, you can wash organs in the sink to clean out the germs on them, but that's not exactly practical. Organ manipulation surgery directly spreading germs is probably rather mean, but I'm not sure how to make hand washing have consequence beyond that. I know! I'll buff spaceacillin a bunch!

**Comment by Crazylemon64** [comment]
4x strength. Better?

**Comment by Plotron** [comment]
Infection in SS13 is like cancer.
You get the news, say goodbye to the world, shoot yourself in the head, and get cloned.

**Comment by Fox-McCloud** [comment]
>You receive the paper, say goodbye to the world, shoot yourself in the head, and get cloned.

This, so much. It's easy to solve external limb infections, but if you have anything beyond a mild infection, internally? It's easier to just kill the person and clone them than to take up the entirety of medbay's spaceacillin supply (I'm not even exaggerating here) for a single person--less time for you and them, too.

**Comment by TheDZD** [comment]
Getting Travis to go through PRs now that it's up and running again.

File: code/modules/organs/organ_external.dm [hunk 1]
```diff
@@ -79,24 +79,28 @@
...
 				return
 		if(1)
 			if(istype(W,/obj/item/weapon/retractor))
+				spread_germs_to_organ(src,user)
 				user.visible_message("<span class='danger'><b>[user]</b> cracks [src] open like an egg with [W]!")
 				stage++
 				return
 		if(2)
 			if(istype(W,/obj/item/weapon/hemostat))
+				spread_germs_to_organ(O,user)
```
Comment (TheDZD) [inline code/modules/organs/organ_external.dm]: Travis exited with an undefined var on this line.

**Comment by Crazylemon64** [comment]
Alright, pinging @tigercat2000, @Aurorablade, and @ZomgPonies in case they might know anything about viewing certain objects with VV causing the client to disconnect

**Comment by Taaketa** [comment]
Could you not program something in like injecting spacillan into the right region of the body with the infection would make it 10x effective?

**Comment by Crazylemon64** [comment]
@Taaketa That would require a nontrivial amount of adjustments to our reagent system first, which is a bit out-of-scope for this PR

**Comment by Fox-McCloud** [comment]
@TheDZD

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 2]
```diff
@@ -12,91 +12,112 @@
...
-		if(istype(O, /obj/item/organ/brain/crystal ))
-			user << "<span class='warning'> This brain is too malformed to be able to use with the [src].</span>"
+/obj/item/device/mmi/attackby(var/obj/item/O as obj, var/mob/user as mob, params)
+	if(istype(O, /obj/item/organ/brain/crystal ))
+		user << "<span class='warning'> This brain is too malformed to be able to use with the [src].</span>"
+		return
+	if(istype(O,/obj/item/organ/brain) && !brainmob) //Time to stick a brain in it --NEO
+		var/obj/item/organ/brain/B = O
+		if(!B.brainmob)
+			user << "\red You aren't sure where this brain came from, but you're pretty sure it's a useless brain."
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Spans.

File: code/modules/mob/living/carbon/brain/MMI.dm [hunk 3]
```diff
@@ -12,91 +12,112 @@
...
-
-		if(brainmob)
-			O.attack(brainmob, user)//Oh noooeeeee
-			// Brainmobs can take damage, but they can't actually die. Maybe should fix.
+		if(B.status & ORGAN_DEAD) // As it stood, a brain would never become unviable for MMI usage. Perhaps we should keep it that way?
+			user << "<span class='warning'>This [B] is dead</span>"
 			return
-		..()
+		for(var/mob/V in viewers(src, null))
+			V.show_message(text("\blue [user] sticks \a [O] into \the [src]."))
```
Comment (TheDZD) [inline code/modules/mob/living/carbon/brain/MMI.dm]: Spans.

**Comment by TheDZD** [comment]
Couldn't find any major issues outside of using color macros instead of text macros. Then again organ/infection code is a horrible mess.

**Comment by Crazylemon64** [comment]
I'll nil the copypasta'd color macros, I guess the eldritch brain stuff is some neglected, ancient aberration in the codebase.

File: code/modules/organs/organ.dm [hunk 4]
```diff
@@ -140,6 +144,32 @@ var/list/organ_cache = list()
...
+			// if the virologist releases gibbingtons again or something
+			// There's probably a much less silly way of doing this, but BYOND native algorithms are stupidly naive
+			is_in_freezer = 0
+			for(var/obj/structure/closet/crate/freezer/F in loc.contents)
+				if(F.opened)
+					is_in_freezer = 1 // on the same tile, close enough, should keep organs much fresher on avg
+					break
+			needs_preservation_update = 0
+			spawn(100) // (100 / 10) once every 10 seconds
+				needs_preservation_update = 1
```
Comment (Crazylemon64) [inline code/modules/organs/organ.dm]: Hrmmm, this should only fire every 10 seconds, is this alright for a spawn() in process()?

**Comment by Crazylemon64** [comment]
Pinging @Fox-McCloud in case the spawn() inside something called by process, used as a cooldown to avoid a naive list walk every frame, is a good thing or a server-freezer.

**Comment by Fox-McCloud** [comment]
sleep() would, as it would tie up the entirety of processing until it was completed.

spawn() spawn it off on its own thread---generally you want to avoid doing that a lot in processing, as it costly, but given taht this doesn't happen a lot....it would still probably be better for this to operate on some type of internal cooldown based on time, where if a threshold time hasn't passed, the rest fails to run...where as if it has passed, then the rest does run--that way you don't have any sleep() or spawn()'s involved in the mess of things.

**Pull request merged**